test/unit/gateways/payflow_test.rb

```diff
@@ -49,6 +49,70 @@ def test_successful_authorization
     refute response.fraud_review?
   end
 
+  def test_successful_purchase_with_stored_credential
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: 'recurring',
+      initiator: 'cardholder',
+      network_transaction_id: nil
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<CardOnFile>CITR</CardOnFile>), data
+    end.respond_with(successful_purchase_with_fraud_review_response)
+
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: 'recurring',
+      initiator: 'merchant',
+      network_transaction_id: '1234'
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<CardOnFile>MITR</CardOnFile>), data
+      assert_match %r(<TxnId>1234</TxnId>), data
+    end.respond_with(successful_purchase_with_fraud_review_response)
+
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: 'unscheduled',
+      initiator: 'merchant',
+      network_transaction_id: '123'
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<CardOnFile>MITU</CardOnFile>), data
+      assert_match %r(<TxnId>123</TxnId>), data
+    end.respond_with(successful_purchase_with_fraud_review_response)
+
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: 'unscheduled',
+      initiator: 'cardholder',
+      network_transaction_id: nil
+    }
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      assert_match %r(<CardOnFile>CITU</CardOnFile>), data
+    end.respond_with(successful_purchase_with_fraud_review_response)
+
+    @options[:stored_credential] = {
+      initial_transaction: true,
+      reason_type: nil,
```

Comment:
We shouldn't have a `reason_type` of `nil`.

Reply:
Updated 👍